src/main/java/org/zeromq/ZContext.java

```diff
@@ -170,9 +182,44 @@ public void destroySocket(Socket s)
         }
     }
 
+    /**
+     * Creates a selector. It needs to be closed by {@link #closeSelector(Selector)}.
+     *
+     * @return a newly created selector.
+     * @deprecated this was exposed by mistake.
+     */
+    @Deprecated
     public Selector createSelector()
     {
-        return context.selector();
+        return selector();
+    }
+
+    /**
+     * Creates a selector. Resource will be released when context will be closed.
+     *
+     * @return a newly created selector.
+     */
+    Selector selector()
+    {
+        Selector selector = context.selector();
+        selectors.add(selector);
+        return selector;
+    }
+
+    /**
+     * Closes a selector.
+     * This is a DRAFT method, and may change without notice.
+     *
+     * @param selector the selector to close. It needs to have been created by {@link #createSelector()}.
+     * @deprecated {@link #createSelector()} was exposed by mistake. while waiting for the API to disappear, this method is provided to allow releasing resources.
+     */
+    @Deprecated
```

Comment:
It feels weird to release a new method that is deprecated straight away. We might want to not deprecate it and only mark it as draft. Thoughts?

Reply:
I'm not sure I understand why we need to add this method, if we don't intend for anyone to use it! :)   IMHO, we should just deprecate  `createSelector`.

Reply:
I start to be lost here by what you want to achieve. Better to make this PR then you can recode what you wish is necessary.

Reply:
So, the problem here, in my mind (and which you captured in #592), is that we expose a method `createSelector`, which is really an implementation detail, since selectors should be managed internally by the context.

A separate issue is that there is no `closeSelector` method, but that's only a problem because `createSelector` exists. Ideally, no one should call `createSelector` because you shouldn't be managing selectors yourself -- you should leave it up to the context.